enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java

```diff
@@ -0,0 +1,201 @@
+     * @param ownerId Owner identification
+     * @throws RemoteCommunicationException In case of remote communication error
+     * @throws DocumentVerificationException In case of business logic error
+     * @throws OnboardingProcessException When process not found
+     */
+    public void executeFinalDocumentVerification(final IdentityVerificationEntity identityVerification, final OwnerId ownerId)
+            throws RemoteCommunicationException, DocumentVerificationException, OnboardingProcessException {
+
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, IN_PROGRESS, ownerId);
+
+        final List<DocumentVerificationEntity> documentVerifications = filterDocumentVerifications(identityVerification);
+
+        final List<String> uploadIds = documentVerifications.stream()
+                .map(DocumentVerificationEntity::getUploadId)
+                .collect(toList());
+
+        final DocumentsVerificationResult result = documentVerificationProvider.verifyDocuments(ownerId, uploadIds);
+        final String verificationId = result.getVerificationId();
+        final DocumentVerificationStatus status = result.getStatus();
+        logger.info("Cross verified documents upload ID: {}, verification ID: {}, status: {}, {}", uploadIds, verificationId, status, ownerId);
+        auditService.auditDocumentVerificationProvider(identityVerification, "Cross verified documents: {} for user: {}", status, ownerId.getUserId());
+
+        documentVerifications.forEach(docVerification -> {
+            docVerification.setVerificationId(result.getVerificationId());
+            docVerification.setTimestampLastUpdated(ownerId.getTimestamp());
+        });
+
+        switch (status) {
+            case ACCEPTED:
+                accept(identityVerification, documentVerifications, ownerId);
+                break;
+            case FAILED:
+                fail(identityVerification, result, documentVerifications, ownerId);
+                break;
+            case REJECTED:
+                reject(identityVerification, result, documentVerifications, ownerId);
+                break;
+            case IN_PROGRESS:
+                throw new DocumentVerificationException("Only sync mode is supported, " + ownerId);
+            default:
+                throw new DocumentVerificationException(String.format("Not supported status %s, %s", status, ownerId));
+        }
+    }
+
+    private void accept(
+            final IdentityVerificationEntity identityVerification,
+            final List<DocumentVerificationEntity> documentVerifications,
+            final OwnerId ownerId) {
+        documentVerifications.forEach(docVerification ->
+            auditService.audit(docVerification, "Document accepted at final verification for user: {}", identityVerification.getUserId()));
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, ACCEPTED, ownerId);
```

Comment:
Not introduced as a separate state of state machine configuration yet. Added just for auditing reasons. For sync processing, this step makes sense to be done all at once. Should be added to the diagram.